### PR TITLE
resourceread: use lib function for loading machine configs

### PR DESCRIPTION
**Pull request by darkmuggle**
Use lib/resourceread for loading machineconfigs for the daemon. Added
basic testing for load function.

### CODE REVIEW

File: pkg/daemon/daemon.go [hunk 1]
```diff
@@ -15,14 +15,13 @@ import (
 	ignv2_2types "github.com/coreos/ignition/config/v2_2/types"
 	"github.com/golang/glog"
 	drain "github.com/openshift/kubernetes-drain"
+	resr "github.com/openshift/machine-config-operator/lib/resourceread"
```
Comment (ashcrow) [inline pkg/daemon/daemon.go]: Note: the rest of the code base uses `resourceread` without an alias, but I'm still OK with this.

File: lib/resourceread/machineconfig.go [hunk 2]
```diff
@@ -17,13 +17,22 @@ func init() {
 	}
 }
 
-// ReadMachineConfigV1OrDie reads MachineConfig object from bytes. Panics on error.
+// ReadMachineConfigV1 reads raw MachineConfig object from bytes. Returns MachineConfig and error.
+func ReadMachineConfigV1(objBytes []byte) (*mcfgv1.MachineConfig, error) {
+	mc, err := runtime.Decode(mcfgCodecs.UniversalDecoder(mcfgv1.SchemeGroupVersion), objBytes)
+	if err == nil {
```
Comment (abhinavdahiya) [inline lib/resourceread/machineconfig.go]:
```go
if err !=nil {
    return nil, err
}
mc, ok := obj.(*mcfgv1.MachineConfig)
if !ok {
    return nil, <error describing this>
}
return mc, nil
```

File: pkg/daemon/daemon.go (hunk 1 above)
Comment (darkmuggle) [reply pkg/daemon/daemon.go]: Fixed

File: lib/resourceread/machineconfig.go (hunk 2 above)
Comment (darkmuggle) [reply lib/resourceread/machineconfig.go]: Thanks for the feedback...I've hardened the logic and repushed.

**Comment by darkmuggle** [comment]
@abhinavdahiya the rationale for `ReadMachineConfigV1` was create a check function that could be used in other parts of the code. Before this commit, individual callers that wanted to parse a MachineConfig had to implement their own logic [1] . This change removes [1] in favor of `ReadMachineConfigV1`, Since `ReadMachineConfigV1OrDie` is used (when a panic is desirable), `ReadMachineConfigV1` handles the parsing of the config for the rest of the code.  

[1]  https://github.com/openshift/machine-config-operator/blob/f915a1bfb2d98a2cdde2ea264008f7dfb58bf262/pkg/daemon/daemon.go#L714-L722

File: lib/resourceread/machineconfig.go [hunk 3]
```diff
@@ -17,13 +19,37 @@ func init() {
...
+	if err != nil {
+		return nil, err
+	}
+	if m == nil {
+		return nil, errDecode
+	}
+
+	mc, ok := m.(*mcfgv1.MachineConfig)
+	if !ok {
+		return nil, errDecode
```
Comment (abhinavdahiya) [inline lib/resourceread/machineconfig.go]: `runtime.Decode(mcfgCodecs.UniversalDecoder(mcfgv1.SchemeGroupVersion), objBytes)` will parse raw bytes of a type in https://github.com/openshift/machine-config-operator/blob/f915a1bfb2d98a2cdde2ea264008f7dfb58bf262/pkg/apis/machineconfiguration.openshift.io/v1/register.go#L40-L49

So this `m.(*mcfgv1.MachineConfig)` is that the raw byte was actually `MachineConfig` object. So the error should be `fmt.Errorf("expected *mcfvgv1.MachineConfig bute found %T", m)`

Comment (darkmuggle) [reply lib/resourceread/machineconfig.go]: Ah, indeed :+1: I've made the error more descriptive and updated the other others too.

**Comment by ashcrow** [comment]
```
could not wait for build: the build machine-config-controller failed after 5m13s with reason DockerBuildFailed: Docker build strategy has failed.
```

Failure to build

/test e2e-aws

**Comment by ashcrow** [comment]
Looks like build issues related to dangling imports.

```
pkg/daemon/daemon.go:25:2: imported and not used: "github.com/openshift/machine-config-operator/vendor/k8s.io/apimachinery/pkg/runtime"
pkg/daemon/daemon.go:26:2: imported and not used: "github.com/openshift/machine-config-operator/vendor/k8s.io/apimachinery/pkg/runtime/serializer"
# github.com/openshift/machine-config-operator/pkg/daemon
pkg/daemon/daemon.go:25:2: imported and not used: "github.com/openshift/machine-config-operator/vendor/k8s.io/apimachinery/pkg/runtime"
pkg/daemon/daemon.go:26:2: imported and not used: "github.com/openshift/machine-config-operator/vendor/k8s.io/apimachinery/pkg/runtime/serializer"
```

**Comment by ashcrow** [comment]
@darkmuggle PTAL at the dangling imports.

**Comment by darkmuggle** [comment]
@ashcrow looks like merge conflicts were the root cause of the dangling imports. I've rebased from master, and pushed. Maybe it'll work now....

**Comment by darkmuggle** [comment]
@abhinavdahiya I believe your requested changes have been addressed? If so, would you mind removing your objection?

**Comment by abhinavdahiya** [comment]
> @abhinavdahiya I believe your requested changes have been addressed? If so, would you mind removing your objection?

Oh, i forgot :innocent: 

/approve

**Pull request merged**